File: internal/model/select_item.go
package model

// SelectItem 选择项（用于前端下拉框、列表选择等）
type SelectItem struct {
	Value     interface{} `json:"value"`
	Label     string      `json:"label"`
	RootId    string      `json:"rootId,omitempty"`
	SubLabels []string    `json:"subLabels,omitempty"`
	LastUse   int64       `json:"lastUse"`
	ExtraData interface{} `json:"extraData,omitempty"`
}

File: frontend/src/model/util/SelectItem.ts
/**
 * 下拉选择框选项
 */
export default class SelectItem {
  value: number | string | null | undefined
  label: string | null | undefined
  subLabels: string[] | null | undefined
  rootId: string | null | undefined
  extraData: object | string | null | undefined

  constructor(selectItem?: SelectItem) {
    if (selectItem === undefined) {
      this.value = undefined
      this.label = undefined
      this.subLabels = undefined
      this.rootId = undefined
      this.extraData = undefined
    } else {
      this.value = selectItem.value
      this.label = selectItem.label
      this.subLabels = selectItem.subLabels
      this.rootId = selectItem.rootId
      this.extraData = selectItem.extraData
    }
  }
}
